boot/boot_serial/src/boot_serial.c

```diff
@@ -68,7 +68,7 @@ BOOT_LOG_MODULE_DECLARE(mcuboot);
 #endif
 
 #define BOOT_SERIAL_INPUT_MAX   512
-#define BOOT_SERIAL_OUT_MAX     (128 * MCUBOOT_IMAGE_NUMBER)
+#define BOOT_SERIAL_OUT_MAX     (128 * BOOT_IMAGE_NUMBER)
```

Comment:
`bootutil_priv.h` also defines BOOT_IMAGE_NUMBER` and performs some validation on it.
Please, consider reusing that definition and removing lines 66-68.

Reply:
right

Reply:
It seems `mcuboot/boot/bootutil/src/` is not part of the include paths for the build of `boot_serial` library, which makes sense since it is not part of `bootutil`.

Does it make sense to move those defines to the public interface of `booutil`?

Reply:
As an alternative you can add `@mcuboot/boot/bootutil` as a dependency here: https://github.com/mcu-tools/mcuboot/blob/main/boot/boot_serial/pkg.yml#L28

Reply:
Oh, I got this wrong, `bootutil_priv.h` is not in an `include/` directory, hmmm, not sure but it does not seem too much like a "private" file after it is shared...

Reply:
Taking a look now my suggestion would be to move:

```
#ifdef MCUBOOT_IMAGE_NUMBER
#define BOOT_IMAGE_NUMBER          MCUBOOT_IMAGE_NUMBER
#else
#define BOOT_IMAGE_NUMBER          1
#endif

_Static_assert(BOOT_IMAGE_NUMBER > 0, "Invalid value for BOOT_IMAGE_NUMBER");
```

to `boot/bootutil/include/bootutil/bootutil.h` so the definitions are not "private" anymore.